### PR TITLE
Add 3003.x and 3004.x to the list of compat versions to test

### PR DESCRIPTION
**Pull request by s0undt3ch**
### What does this PR do?
See title

### CODE REVIEW

File: tests/pytests/scenarios/compat/test_with_versions.py [hunk 1]
```diff
@@ -53,13 +53,13 @@ def _get_test_versions():
             test_versions.append(
                 PySaltCombo(python_version=python_version, salt_version=salt_version)
             )
-    for salt_version in ("3001.4", "3002.2"):
+    for salt_version in ("3001.0", "3002.0", "3003.0", "3004.0"):
```
Comment (Ch3LL) [inline tests/pytests/scenarios/compat/test_with_versions.py:56]: 3001 is out of support. Can we remove that version now?

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: tests/pytests/scenarios/compat/test_with_versions.py (hunk 1 above)
Comment (s0undt3ch) [reply tests/pytests/scenarios/compat/test_with_versions.py:56]: Of course!
How about the other even older than that?
Nuke those too?

Comment (Ch3LL) [reply tests/pytests/scenarios/compat/test_with_versions.py:56]: yep :)

Comment (s0undt3ch) [reply tests/pytests/scenarios/compat/test_with_versions.py]: Fixed

**Comment by s0undt3ch** [comment]
Tests are passing, merging.

**Review by krionbsd — APPROVED** [review]

**Pull request merged**